.github/workflows/create-test-matrices/action.yml

```diff
@@ -0,0 +1,71 @@
+inputs:
+    language-name:
+        description: "Language name"
+        required: true
+        type: choice
+        options:
+            - java
+            - node
+            - python
+            - go
+            - C#
+    run-full-matrix:
+        description: "Run the full matrix"
+        required: true
+        type: boolean
+
+outputs:
+    engine-matrix-output:
+        description: "Engine matrix"
+        value: ${{ steps.load-engine-matrix.outputs.engine-matrix }}
+    host-matrix-output:
+        description: "Host matrix"
+        value: ${{ steps.load-host-matrix.outputs.host-matrix }}
+    version-matrix-output:
+        description: "Version matrix"
+        value: ${{ steps.create-lang-version-matrix.outputs.version-matrix }}
+
+runs:
+    using: "composite"
+    steps:
+        - name: Load engine matrix
+          id: load-engine-matrix
+          shell: bash
+          run: |
+              echo 'Select server engines to run tests against'
+              if [[ "${{ github.event_name }}" == "pull_request" || "${{ github.event_name }}" == "push" || "${{ inputs.run-full-matrix }}" == "false" ]]; then
+                  echo 'Pick engines marked as `"run": "always"` only - on PR, push or manually triggered job which does not require full matrix'
+                  echo engine-matrix=$(jq -c '[.[] | select(.run == "always")]' < .github/json_matrices/engine-matrix.json) >> $GITHUB_OUTPUT
+              else
+                  echo 'Pick all engines - on cron (schedule) or if manually triggered job requires a full matrix'
+                  echo engine-matrix=$(jq -c . < .github/json_matrices/engine-matrix.json) >> $GITHUB_OUTPUT
+              fi
+              cat $GITHUB_OUTPUT
```

Comment:
Good for debugging, but might be also too noisy, I'm not sure what is my opinion here, just consider it.

Plus — wouldn't each step will include the whole out put from step happened before?
Maybe it's better to have a finalize step to cat it all at once?